sql/core/src/test/resources/sql-tests/inputs/datetime.sql

```diff
@@ -75,3 +75,28 @@ select date '2001-09-28' - null;
 -- subtract dates
 select null - date '2019-10-06';
 select date '2001-10-01' - date '2001-09-28';
+
+-- variable-length tests
+select to_timestamp(v, 'yyyy-MM-dd HH:mm:ss.SSSSSS[zzz]') from values
+ ('2019-10-06 10:11:12.'),
```

Comment:
is it legal? Have you tried it with Spark 2.4?

Reply:
I guess this is fine even though 2.4 will return `null` with `yyyy-MM-dd HH:mm:ss.S` but success with `yyyy-MM-dd HH:mm:ss.S`. And also it will fail to parse when I set `S` to optional, this looks weird to me.

Here the `.` in java 8 time API is JUST a separator, it is not the decimal point, we can put anything after it, eg. timezone, week, `.` e.t.c.

Reply:
maybe we should require at least one number for `SSS`?

Reply:
sgtm

Reply:
1 looks better since it fits better both for 2.4 and the current behavior in 3.x. Now we have SSS means [1, 3] numbers, and [SSS] means[0, 3] numbers